src/viewProviders/flinkArtifacts.test.ts

```diff
@@ -0,0 +1,179 @@
+      sinon.assert.calledWith(stubbedLoader.getFlinkArtifacts, resource);
+      assert.deepStrictEqual(viewProvider["_artifacts"], mockArtifacts);
+    });
+
+    it("should handle 401 auth errors and fire auth session invalidated", async () => {
+      const windowWithProgressStub = sandbox
+        .stub(window, "withProgress")
+        .callsFake((_, callback) => {
+          // Call the callback immediately with a resolved promise
+          const mockProgress = {} as Progress<unknown>;
+          const mockToken = {} as CancellationToken;
+          return Promise.resolve(callback(mockProgress, mockToken));
+        });
+
+      const authInvalidatedFireStub = sandbox.stub(ccloudAuthSessionInvalidated, "fire");
+      const resource = TEST_CCLOUD_FLINK_COMPUTE_POOL;
+      viewProvider["resource"] = resource;
+
+      const stubbedLoader: sinon.SinonStubbedInstance<CCloudResourceLoader> =
+        getStubbedCCloudResourceLoader(sandbox);
+
+      const authError = createResponseError(401, "Unauthorized", "test");
+      stubbedLoader.getFlinkArtifacts.rejects(authError);
+
+      await assert.rejects(async () => {
+        await viewProvider.refresh();
+      });
+
+      sinon.assert.calledOnce(windowWithProgressStub);
+      sinon.assert.calledOnce(stubbedLoader.getFlinkArtifacts);
+      sinon.assert.calledOnce(authInvalidatedFireStub);
+      assert.deepStrictEqual(viewProvider["_artifacts"], []);
+    });
+
+    it("should handle non-401 errors without firing auth session invalidated", async () => {
+      const windowWithProgressStub = sandbox
+        .stub(window, "withProgress")
+        .callsFake((_, callback) => {
+          // Call the callback immediately with a resolved promise
+          const mockProgress = {} as Progress<unknown>;
+          const mockToken = {} as CancellationToken;
+          return Promise.resolve(callback(mockProgress, mockToken));
+        });
+
+      const authInvalidatedFireStub = sandbox.stub(ccloudAuthSessionInvalidated, "fire");
+      const resource = TEST_CCLOUD_FLINK_COMPUTE_POOL;
+      viewProvider["resource"] = resource;
+
+      const stubbedLoader: sinon.SinonStubbedInstance<CCloudResourceLoader> =
+        getStubbedCCloudResourceLoader(sandbox);
+
+      const serverError = createResponseError(500, "Internal Server Error", "test");
+      stubbedLoader.getFlinkArtifacts.rejects(serverError);
+
+      await assert.rejects(async () => {
+        await viewProvider.refresh();
+      });
+
+      sinon.assert.calledOnce(windowWithProgressStub);
+      sinon.assert.calledOnce(stubbedLoader.getFlinkArtifacts);
+      sinon.assert.notCalled(authInvalidatedFireStub);
+      assert.deepStrictEqual(viewProvider["_artifacts"], []);
+    });
+  });
+
+  it("returns artifacts when compute pool is selected", () => {
+    const mockArtifacts = [
+      new FlinkArtifact({
+        connectionId: TEST_CCLOUD_FLINK_COMPUTE_POOL.connectionId,
+        connectionType: "ccloud" as ConnectionType.Ccloud,
```

Comment:
One other minor update, since it's technically [`"CCLOUD"`](https://github.com/confluentinc/vscode/blob/9474c1448f14d90839c8367cf79957dfe21a590b/src/clients/sidecar/models/ConnectionType.ts#L20-L22):
```suggestion
        connectionType: ConnectionType.Ccloud,
```